test/models/concerns/smart_proxy_extensions_test.rb

```diff
@@ -69,6 +71,21 @@ def test_rhsm_url_pulp_primary
     def test_rhsm_url_pulp_mirror
       assert_includes @proxy_mirror.rhsm_url, ":8443/rhsm"
     end
+
+    def test_sync_container_gateway
+      environment = katello_environments(:library)
+
+      with_pulp3_features(capsule_content.smart_proxy)
+      capsule_content.smart_proxy.add_lifecycle_environment(environment)
+
+      capsule_content.smart_proxy.expects(:update_container_repo_list).returns(true)
+      capsule_content.smart_proxy.expects(:update_user_container_repo_mapping).returns(true)
+      capsule_content.smart_proxy.expects(:container_gateway_users).returns(::User.where(login: 'secret_admin'))
+
+      capsule_content.smart_proxy.update_container_repo_list
+      users = capsule_content.smart_proxy.update_user_container_repo_mapping
```

Comment:
`users` should be fetched from `smart_proxy.container_gateway_users`.  That won't matter though if you run `sync_container_gateway` directly instead of running its individual methods.